src/gen9_vdenc.c

```diff
@@ -3670,13 +3670,13 @@ gen9_vdenc_avc_encode_picture(VADriverContextP ctx,
         gen9_vdenc_mfx_vdenc_pipeline(ctx, encode_state, encoder_context);
         gen9_vdenc_read_status(ctx, encoder_context);
 
-        intel_batchbuffer_end_atomic(batch);
-        intel_batchbuffer_flush(batch);
-
         vdenc_context->brc_initted = 1;
         vdenc_context->brc_need_reset = 0;
     }
 
+    intel_batchbuffer_end_atomic(batch);
+    intel_batchbuffer_flush(batch);
+
     return VA_STATUS_SUCCESS;
 }
 
```

Comment:
The main purpose is to avoid the issue that the status register in previous is not stored while it is used in the subsequent pass.  It seems that the current code already stores the register into the internal buffer. In such case the commit log is incorrect.

Of course the patch is to avoid submitting batch buffer multi times. When it works in multi-pass scenarios, maybe the first pass already meets with the requirement. In such case subsequent pass is not necessary.  But the current code still prepares it and submits it. It is not meaningful.